client/my-sites/site-settings/wrap-settings-form.jsx

```diff
@@ -108,14 +108,20 @@ const wrapSettingsForm = getFormSettings => SettingsForm => {
 			this.props.trackEvent( 'Clicked Save Settings Button' );
 		};
 
-		submitForm = () => {
+		submitForm = ( callback ) => {
 			const { fields, settingsFields, siteId, jetpackSettingsUISupported } = this.props;
 			this.props.removeNotice( 'site-settings-save' );
 
-			this.props.saveSiteSettings( siteId, pick( fields, settingsFields.site ) );
+			const settingsRequests = [
+				this.props.saveSiteSettings( siteId, pick( fields, settingsFields.site ) )
+			];
+
 			if ( jetpackSettingsUISupported ) {
-				this.props.updateSettings( siteId, pick( fields, settingsFields.jetpack ) );
+				settingsRequests.push( this.props.updateSettings( siteId, pick( fields, settingsFields.jetpack ) ) );
 			}
+
+			Promise.all( settingsRequests )
```

Comment:
What about returning this promise instead of using a callback pattern?

The callback pattern is used on some methods here because `setState()` takes a callback arg, due to an async behaviour when updating the component's state but this is not the case here. 

Also take in account that this callback or the resolution of the Promise would be executed even when the request fails. This is not a problem here, because an extra request on failure wouldn't do any harm.